plugin.video.haystacktv/resources/lib/haystack.py

```diff
@@ -0,0 +1,238 @@
+#   Copyright (C) 2017 Lunatixz
+#
+#
+# This file is part of Haystack.TV.
+#
+# Haystack.TV is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Haystack.TV is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Haystack.TV.  If not, see <http://www.gnu.org/licenses/>.
+
+# -*- coding: utf-8 -*-
+import os, sys, datetime, re, traceback
+import urllib, urllib2, socket, json
+import xbmc, xbmcgui, xbmcplugin, xbmcaddon
+
+from simplecache import SimpleCache
+from bs4 import BeautifulSoup
+from YDStreamExtractor import getVideoInfo
+
+# Plugin Info
+ADDON_ID      = 'plugin.video.haystacktv'
+REAL_SETTINGS = xbmcaddon.Addon(id=ADDON_ID)
+ADDON_NAME    = REAL_SETTINGS.getAddonInfo('name')
+SETTINGS_LOC  = REAL_SETTINGS.getAddonInfo('profile')
+ADDON_PATH    = REAL_SETTINGS.getAddonInfo('path').decode('utf-8')
+ADDON_VERSION = REAL_SETTINGS.getAddonInfo('version')
+ICON          = REAL_SETTINGS.getAddonInfo('icon')
+FANART        = REAL_SETTINGS.getAddonInfo('fanart')
+LANGUAGE      = REAL_SETTINGS.getLocalizedString
+
+## GLOBALS ##
+TIMEOUT       = 15
+CONTENT_TYPE  = 'files'
+DEBUG         = REAL_SETTINGS.getSetting('Enable_Debugging') == 'true'
+BASEURL       = 'http://www.haystack.tv'
+YTURL         = 'plugin://plugin.video.youtube/play/?video_id='
+VMURL         = 'plugin://plugin.video.vimeo/play/?video_id='
+IGNORE        = ', watch it on the web, Android, iPad and Chromecast.'
+# MY HEADLINES,EDITOR'S PICKS,NEWS,SCITECH,ENTERTAINMENT,BUSINESS,U.S.POLITICS,SPORTS,GAMING,LATE NIGHT,MOVIES,POPULAR
+
+def log(msg, level=xbmc.LOGDEBUG):
+    if DEBUG == False and level != xbmc.LOGERROR: return
+    if level == xbmc.LOGERROR: msg += ' ,' + traceback.format_exc()
+    xbmc.log(ADDON_ID + '-' + ADDON_VERSION + '-' + msg, level)
+    
+def getParams():
+    param=[]
+    if len(sys.argv[2])>=2:
+        params=sys.argv[2]
+        cleanedparams=params.replace('?','')
+        if (params[len(params)-1]=='/'): params=params[0:len(params)-2]
+        pairsofparams=cleanedparams.split('&')
+        param={}
+        for i in range(len(pairsofparams)):
+            splitparams={}
+            splitparams=pairsofparams[i].split('=')
+            if (len(splitparams))==2: param[splitparams[0]]=splitparams[1]
+    return param
```

Comment:
The following one-liner is enough for simple query strings (without repeating keys):
```python
param = dict(urlparse.parse_qsl(sys.argv[2][1:]))
```

Reply:
Thanks, I'll look into the change.

Reply:
Accidentally closed the branch when updating recommend changes... I'll rebase.